### PR TITLE
#438 Contexts share options bugfixes, icon update

### PR DESCRIPTION
**Pull request by alexander-fedorenko**
## Description
- Support a way to disable share menu for contexts tab too.
- Use default share options for Context category if FeaturedMaps doesn't overwrite these settings.
- Introduces new icon for contexts in featured items grid; for contexts tab when no context created.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

[#445](https://github.com/georchestra/mapstore2-georchestra/pull/445#issuecomment-1014702806)

**What is the new behavior?**
"Embed" tab is properly displayed in share options modal for featured Context
Share tool can be disabled for "Contexts" tab at home page (similar to previously introduced for Maps/GeoStories/Dashboards #7624)

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by tdipisa** [comment]
@offtherailz please revire this asap tomorrow so that @alexander-fedorenko can update the client project and we can proceed with final tests in general.

File: web/client/utils/ShareUtils.js [hunk 1]
```diff
@@ -36,6 +36,18 @@ export const GEOSTORY_DEFAULT_SHARE_OPTIONS = {
     }
 };
 
+export const CONTEXT_DEFAULT_SHARE_OPTIONS = {
+    embedPanel: true,
+    showAPI: false,
+    embedOptions: { showTOCToggle: false, allowFullScreen: false },
+    shareUrlRegex: "(h[^#]*)#\\/context\\/([^\\/]*)\\/([A-Za-z0-9]*)",
+    shareUrlReplaceString: "$1context-embedded.html#/$3",
```
Comment (offtherailz) [inline web/client/utils/ShareUtils.js:44]: `contexts-embedded.html` doesn't exist and it doesn't seems to be used (is it?). I think we can remove this option

File: web/client/themes/default/icons.less [hunk 2]
```diff
@@ -581,152 +582,153 @@ structure .glyphicon-{iconName}-content
...
-.glyphicon-usd-content { content: "\f24c"; }
-.glyphicon-user-content { content: "\f24d"; }
-.glyphicon-vert-dashed-content { content: "\f24e"; }
-.glyphicon-warning-sign-content { content: "\f24f"; }
-.glyphicon-webpage-content { content: "\f250"; }
-.glyphicon-wrench-content { content: "\f251"; }
-.glyphicon-zoom-in-content { content: "\f252"; }
-.glyphicon-zoom-out-content { content: "\f253"; }
-.glyphicon-zoom-to-content { content: "\f254"; }
+.glyphicon-map-context-content { content: "\f1c0"; }
```
Comment (offtherailz) [inline web/client/themes/default/icons.less:585]: Looks strange to me that all the codes has been re-mapped. @allyoucanmap 
Is it the normal behavior of the script? In the past I've seen only a line added.

**Review by offtherailz — CHANGES_REQUESTED** [review]
Generally speaking everything works well. Just a few comments for clean up and a clarification about icons that I ask also to @allyoucanmap

File: web/client/themes/default/icons.less (hunk 2 above)
Comment (offtherailz) [reply web/client/themes/default/icons.less:585]: @alexander-fedorenko , generation looks ok. Anyway I can not see the new svg in the folder. It have to be added too, in order to be present in future generations of icons.

Comment (alexander-fedorenko) [reply web/client/themes/default/icons.less:585]: Thank, added svg file to the tree.

**Review by offtherailz — APPROVED** [review]

**Pull request merged**